.github/workflows/golangci.yml

```diff
@@ -7,7 +7,15 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+          persist-credentials: false
+          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```

Comment:
What exactly is the goal of this configuration?

Reply:
My understanding of the action definition here is that when opening a PR, we're going to checkout the repository. Telling the action that the `ref` is `main` will force the checkout of the `main` branch, and the branch that we would like to build for the open PR.
Am I mistaken ?

Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

Reply:
> Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

I am not sure I understand that. My understanding is that the github action needs to download a private repo and it therefore needs the appropriate credentials for that.

Reply:
> Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

I am not sure I understand that. My understanding is that the github action needs to download a private repo and it therefore needs the appropriate credentials for that.

> Telling the action that the ref is main will force the checkout of the main branch, and the branch that we would like to build for the open PR.

Yeah... I kinda copy-pasted things there. I am not sure of what I did. Would you suggest to remove the `ref` key or to set it to [.github.ref](https://docs.github.com/en/actions/learn-github-actions/variables#:~:text=processing%20using%20the-,%24%7B%7B%20github.ref%20%7D%7D,-context%20property.)? I'll try things out [there](https://github.com/edgelaboratories/daycount/pull/65)

Reply:
As discussed with @olivierjacob, it seems that daycount does to have access to the saem secrets than other repos. I'll clean up the history here and will create a new PR for fixing the secrets and will ping the SREs on that other PR.